src/components/Design.vue

```diff
@@ -0,0 +1,341 @@
+                <v-layout column>
+                  <v-flex xs4>Select a reaction source database</v-flex>
+                  <v-checkbox
+                    v-model="bigg"
+                    label="BiGG"
+                    color="primary"
+                  ></v-checkbox>
+                  <v-checkbox
+                    v-model="rhea"
+                    label="Rhea"
+                    color="primary"
+                  ></v-checkbox>
+                </v-layout>
+                <v-autocomplete
+                  label="Model"
+                  v-model="model"
+                  :items="modelOptions"
+                  item-text="name"
+                  item-value="id"
+                  return-object
+                  :rules="modelRules()"
+                  clearable
+                >
+                  <template v-slot:prepend-item>
+                    <v-btn
+                      depressed
+                      @click.stop="isModelCreationDialogVisible = true"
+                    >
+                      <v-icon class="mr-4">add_circle</v-icon>
+                      New model
+                    </v-btn>
+                    <v-divider></v-divider>
+                  </template>
+                </v-autocomplete>
+                <v-text-field
+                  label="Maximum number of pathway predictions"
+                  v-model="maxPredictions"
+                  class="mt-0"
+                  :rules="predictionRules"
+                  type="number"
+                ></v-text-field>
+              </v-card-text>
+            </v-slide-y-transition>
+          </v-card>
+          <v-btn class="primary" :disabled="!isValid" @click="onSubmit"
+            >Submit
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+    <v-snackbar color="error" v-model="hasSubmissionError" :timeout="8000">
+      Sorry, submitting your design job failed. Please try again in a few
+      seconds or contact us if the problem persists.
+    </v-snackbar>
+  </v-container>
+</template>
+
+<script lang="ts">
+import Vue from "vue";
+import { Route, RawLocation } from "vue-router";
+import axios, { AxiosResponse } from "axios";
+import * as settings from "@/settings";
+import { OrganismItem } from "@/store/modules/organisms";
+import { ProjectItem } from "@/store/modules/projects";
+import { ModelItem, organism2ModelMapping } from "@/store/modules/models";
+import NewProject from "./shared/NewProject.vue";
+import NewOrganism from "./shared/NewOrganism.vue";
+import NewModel from "./shared/NewModel.vue";
+
+interface ProductItem {
+  name: string;
+}
+
+type Nullable<T> = T | null | undefined;
+type RuleOutcome = string | true;
+type RuleHandler = (any) => RuleOutcome;
+type NextHandler = (to?: RawLocation | false | Function | void) => void;
+
+export default Vue.extend({
+  name: "Design",
+  components: {
+    NewProject,
+    NewOrganism,
+    NewModel
+  },
+  data() {
+    return {
```

Comment:
Okay, that was supposed to be the point :frowning_face: maybe that means I need to define an interface and let `data()` return that?

Reply:
Alright, I have tried again with defining an interface. On one hand it's still possible to assign any other types which probably has to do with the component state in practice being something else than the object returned by `data()`. On the other hand, having the interface definition still makes auto-completion on properties possible so that's quite nice. I'd call it a nice-to-have. Additionally, defining an interface for `data()` also documents the intention for the various data types.